pkg/httpserver/traffic_redistribution_middleware.go

```diff
@@ -0,0 +1,152 @@
+package httpserver
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/justtrackio/gosoline/pkg/appctx"
+	"github.com/justtrackio/gosoline/pkg/cache"
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/clock"
+	"github.com/justtrackio/gosoline/pkg/log"
+)
+
+const (
+	distributorMaxSize   int64  = 10000
+	distributorPruneSize uint32 = 1000
+)
+
+//go:generate mockery --name TrafficDistributor
+type (
+	TrafficDistributor interface {
```

Comment:
I dislike the name because it doesn't distribute traffic, it closes connections, so it is more a connection manager (but manager itself is an empty word) or connection limiter or connection closer than a traffic distributor